oauth-common/src/main/java/io/strimzi/kafka/oauth/validator/OAuthIntrospectionValidator.java

```diff
@@ -272,16 +273,14 @@ public TokenInfo validate(String token) {
 
         } catch (IOException e) {
             addIntrospectHttpMetricErrorTime(e, requestStartTime);
-            throw new RuntimeException("Failed to introspect token - send, fetch or parse failed: ", e);
+            throw new ValidationException("Failed to introspect token - send, fetch or parse failed: ", e);
         }
 
-        boolean active;
-        try {
-            active = response.get("active").asBoolean();
-        } catch (Exception e) {
-            throw new RuntimeException("Failed to introspect token - invalid response: \"active\" attribute is missing or not a boolean (" + response.get("active") + ")");
+        JsonNode activeAttr = response.get("active");
+        if (!(activeAttr instanceof BooleanNode)) {
+            throw new ValidationException("Failed to introspect token - invalid response: \"active\" attribute is missing or not a boolean (" + activeAttr + ")");
```

Comment:
Maybe we could log the response too, WDYT?

Reply:
Not sure about that. My concern is that some info that should not enter the log files may be part of the response.

The root of the problem will ultimately be a misconfigured or incompatible authorization server. But if by any chance it includes the raw token in the response and we log it, it's possible that the token is actually valid, but the response contains something like "valid": "true" rather than "active": true. We would log it as error, and if it contains the access token (it's not supposed to, but if the server is some bespoke in-house thing all is possible) the log collection may insert it into some central log system, and there you have it in the clear a valid access token that is possibly usable on some other Kafka instance (maybe one configured with JWT signature check, rather than with introspection validation).

Reply:
Actually you can log all HTTP responses already by setting `io.strimzi.kafka.oauth.common.HttpUtil` Logger level to `TRACE`.
For development that should be enough already to get to the bottom of the problem here.